### ISSUE TITLE
UP031 does not support %d formatting

### ISSUE THREAD

**Issue by inoa-jboliveira** [astral-sh/ruff#12461]
The following conversion codes are currently being ignored by UP031 instead of showing as errors:

```
print("%d" % 1)
print("%i" % 1)
print("%u" % 1)
print("%c" % 1) 
```

The presence of any of these items such as `%d` in a string will make it ignored by UP031 even if it has another supported one, such as `%s`:

```
print("%d - %s" % (1, 2))
```

Please support all of the valid conversion codes from this table: https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting

Also, if an invalid code is present (either by you not supporting or being invalid), still it should show the error for valid ones. The only difference is that you don't want to supply a fix in this case

**Comment by inoa-jboliveira** [comment]
Giving a quick look at the implementation, all this sounds like it is by design. 
There is nothing in the docs that say otherwise, but [I see a test file](https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/resources/test/fixtures/pyupgrade/UP031_1.py) that have several formatting lines are probably being ignored.

It does sound that these options are being accepted only because there is no automated fix instead of because they are fine. There is zero reason for someone to write this in 2024 (or even 10 years ago), as the test file suggests:

```
"%(1)s" % {"1": "bar"}
```

All the old style string formatting options, except for bytestrings, should be deemed errors under UP031. Having a fix available for those cases or not is what should be considered instead.

**Comment by dylwil3** [comment]
This appears to have been fixed by #11229, in preview mode at the time of this writing- at least, I can no longer reproduce the "All checks pass!"

[Playground link](https://play.ruff.rs/9b318e0c-ecbc-4f09-8c1a-f24ed7e08edc)

**Comment by inoa-jboliveira** [comment]
I still get the same behavior with latest ruff

```python
$ cat foo.py
print("%d" % 1)
print("%i" % 1)
print("%u" % 1)
print("%c" % 1)

$ ruff check --select UP --fixable UP --unsafe-fixes --isolated foo.py
All checks passed!

$ ruff version
ruff 0.7.4 (ed7b98cf9 2024-11-15)
```

**Comment by inoa-jboliveira** [comment]
Ohh sorry!! I forgot to add the "--preview". 
Yes, seem like this issue can be closed then

**Comment by dylwil3** [comment]
Hmm - would you mind trying one more time but with the preview flag `--preview`? Apologies if I'm getting something backwards here

**Comment by inoa-jboliveira** [comment]
Yes, thank you!
